Userland/Services/WindowServer/WindowManager.cpp

```diff
@@ -1960,21 +1960,62 @@ ResizeDirection WindowManager::resize_direction_of_window(Window const& window)
     return m_resize_direction;
 }
 
-Gfx::IntRect WindowManager::maximized_window_rect(Window const& window, bool relative_to_window_screen) const
+Gfx::IntRect WindowManager::maximized_window_rect(Window const& window, bool relative_to_window_screen, WindowTileType tile_type) const
 {
     auto& screen = Screen::closest_to_rect(window.frame().rect());
     Gfx::IntRect rect = screen.rect();
 
+    if (tile_type == WindowTileType::Left
+        || tile_type == WindowTileType::TopLeft
+        || tile_type == WindowTileType::BottomLeft) {
+        rect.set_width(rect.width() / 2);
+    }
+
+    if (tile_type == WindowTileType::Right
+        || tile_type == WindowTileType::TopRight
+        || tile_type == WindowTileType::BottomRight) {
+        rect.set_width(rect.width() / 2);
+        rect.set_x(rect.width());
+    }
+
+    if (tile_type == WindowTileType::Top
+        || tile_type == WindowTileType::TopLeft
+        || tile_type == WindowTileType::TopRight) {
+        rect.set_height(rect.height() / 2);
+    }
+
+    if (tile_type == WindowTileType::Bottom
+        || tile_type == WindowTileType::BottomLeft
+        || tile_type == WindowTileType::BottomRight) {
+        rect.set_height(rect.height() / 2);
+        rect.set_y(rect.height());
+    }
```

Comment:
If rect is odd there will be a gap between bottom tiles and the bar. Can't get away from that tasteful pixel yet^
```
auto taskbar_gap = rect.height() % 2;
rect.set_height(rect.height() / 2 + taskbar_gap);
rect.set_y(rect.height() - taskbar_gap)
```